moneysnake/financial_statement.py

```diff
@@ -18,6 +19,32 @@ class FinancialStatement(MoneybirdModel):
     importer_service: Optional[str] = None
     financial_mutations: List[FinancialMutation] = field(default_factory=list)
 
+    def save(self) -> None:
+        """
+        Save the external sales invoice. Overrides the save method in MoneybirdModel.
+        """
+        financial_statement_data = self.to_dict()
+        # For the POST and PATCH requests we need to use the details_attributes key
+        # instead of details key to match the Moneybird API.
+        financial_statement_data["financial_mutation_attributes"] = (
+            financial_statement_data.pop("financial_mutations", [])
+        )
+
+        if self.id is None:
+            data = post_request(
+                f"{self.endpoint}s",
+                data={self.endpoint: financial_statement_data},
+                method="post",
+            )
+            self.update(data)
```

Comment:
You can move the self.update to after the if-else statement, you might need to define the data variable before the if-else statement with proper typing (If strict is applied)

Reply:
Updated!